snapshot/pkg/volume/openebs/processor.go

```diff
@@ -211,11 +212,14 @@ func (h *openEBSPlugin) SnapshotRestore(snapshotData *crdv1.VolumeSnapshotData,
 	// restore snapshot to a PV
 	snapshotID := snapshotData.Spec.OpenEBSSnapshot.SnapshotID
 	pvRefName := snapshotData.Spec.PersistentVolumeRef.Name
-	pvRefNamespace := snapshotData.Spec.PersistentVolumeRef.Namespace
```

Comment:
who is supposed to set this reference?

Reply:
But Namespace field is not beine g populated by the controller for right now, 
The snapshot creation start with unique name ( <namespace>/snapshot-name>) combination of namespace and snapshotName, which is getting used in all the places and at the end once snapshot create successful the same unique name is added in VolumeSnapshotRef.Name field.

Reply:
But Namespace field is not beine g populated by the controller for right now, 
The snapshot creation start with unique name ( <namespace>/snapshot-name>) combination of namespace and snapshotName, which is getting used in all the places and at the end once snapshot create successful the same unique name is added in VolumeSnapshotRef.Name field.

`VolumeSnapshotRef.Name` are using here to get the namespace of snapshot by spilting them from the unique name 
which is combination of `Namespace/snapshotName`